spec_01.md

```diff
@@ -0,0 +1,46 @@
+- Spec: 1
+- Title: Methods on libraries
+- Version: 0.1
+- Last-Modified: 2019-07-19
+- Authors: [James Bednar](),
+           [Julia Signell](jsignell@gmail.com),
+           [Jake Vanderplas]()
+- Status: Active
+- Type: Standards
+- Content-Type: text/markdown
+- Created: 2019-07-19
+
+## Abstract
+This spec describes a proposed shared API for enabling plotting libraries.
+
+This spec uses LIBRARY to refer to any specific visualization library.
+
+## If a method doesn't make sense
+If a given operation doesn't make sense for that library, then it can satisfy the spec by simply having that method return a message to that effect ("JSON support not available in LIBRARY").
+
+## `.__spec_version__()` method
+Calling `.__spec_version__()` on the library should return a list of the specs that the library complies with. It should include the version of the spec that is being used.
+
+```python
+>>> LIBRARY.__spec_version__()
```

Comment:
Could this be made more explicit? Something like `__pyviz_version__` or `__pyviz_plot_version__` (oh that gets long fast)? Main concerns are `spec` being too generic and future specifications by this group (plotting versus maps versus jupyter widget interface versus something else). Maybe they are all in one large specification, but want to point it out at least.

Reply:
maybe `__pyviz_spec_version__`?

Reply:
Sounds good to me.  Good point, @djhoese .

Reply:
Or maybe just `__pyviz_spec__`.